client/packages/coldchain/src/Equipment/ImportAsset/EquipmentImportModal.tsx

```diff
@@ -133,29 +135,36 @@ export const EquipmentImportModal: FC<EquipmentImportModalProps> = ({
     FileUtils.exportCSV(csv, t('filename.cce-failed-uploads'));
     success(t('success'))();
   };
+  const importErrorRows: ImportRow[] = [];
+  const insertAsset = async (row: ImportRow) => {
+    try {
+      await insertAssets(toInsertEquipmentInput(row, catalogueItemData?.nodes));
+      await insertLog({
+        id: FnUtils.generateUUID(),
+        assetId: row.id,
+        comment: t('label.created'),
+        status: AssetLogStatusInput.Functioning,
+      });
+    } catch (e) {
+      const errorMessage = (e as Error).message ?? t('messages.unknown-error');
+
+      importErrorRows.push({
+        ...row,
+        errorMessage,
+      });
+    }
+  };
 
   const importAction = async () => {
     onChangeTab(Tabs.Import);
     const numberImportRecords = bufferedEquipment?.length ?? 0;
     if (bufferedEquipment && numberImportRecords > 0) {
-      const importErrorRows: ImportRow[] = [];
+      importErrorRows.length = 0;
       // Import count can be quite large, we break this into blocks of 100 to avoid too much concurency
```

Comment:
BTW I think this 100 comes from Health Supply Hub, we put a high number in here to make it faster, but I think have less outstanding concurrent requests is probably better?

The main performance issue we had (which I think we have here too) is that after each promise resolves React re-renders and re-queries the List View in the background.

There's an issue to fix this in HSH here...
https://github.com/msupply-foundation/health-supply-hub/issues/993

Here's an example how we used invalidateQueries after the import finishes in Universal Codes...

https://github.com/msupply-foundation/unified-codes/blob/59ff94c390aa55468a26d10ae2167bbfb5d2f849/frontend/common/src/ui/components/toolbars/SearchAndDeleteToolbar.tsx#L68

Reply:
cool - thanks James. Have refactored to only invalidate after all inserts are done